plugins/generator-1.20.4/neoforge-1.20.4/templates/elementinits/enchantments.java.ftl

```diff
@@ -38,10 +38,10 @@ package ${package}.init;
 
 public class ${JavaModName}Enchantments {
 
-	public static final DeferredRegister<Enchantment> REGISTRY = DeferredRegister.create(ForgeRegistries.ENCHANTMENTS, ${JavaModName}.MODID);
+	public static final DeferredRegister<Enchantment> REGISTRY = DeferredRegister.create(Registries.ENCHANTMENT, ${JavaModName}.MODID);
 
 	<#list enchantments as enchantment>
-	public static final RegistryObject<Enchantment> ${enchantment.getModElement().getRegistryNameUpper()} =
+	public static final DeferredHolder<Enchantment, ${enchantment.getModElement().getName()}Enchantment> ${enchantment.getModElement().getRegistryNameUpper()} =

```

Comment:
Do we need specific class name here?

![image](https://github.com/MCreator/MCreator/assets/16374228/3a8dcf07-032d-49c5-a797-67bdf6bdc12d)

Potions seem to work fine without it. It optimizes generator performance as there is one insertion token in global file less

Reply:
NeoForge used the specific class name i  their example, so I thought it was the best thing to do.

Reply:
Neither do I, but I suppose it is to return the right type when getting an instance. However, this is not really something we need so for the sake of shorter code and faster FTL we use parent object for both